src/transformers/pipelines/base.py

```diff
@@ -969,7 +969,8 @@ def torch_dtype(self) -> Optional["torch.dtype"]:
         """
         Torch dtype of the model (if it's Pytorch model), `None` otherwise.
         """
-        return getattr(self.model, "dtype", None)
+        model_dtype = getattr(self.model, "dtype", None)
+        return model_dtype if is_torch_available and isinstance(model_dtype, torch.dtype) else None
```

Comment:
```suggestion
        return model_dtype if  "torch" in  type(model_dtype) else None
```
might be less costly?